lib/src/core/better_player_controller.dart

```diff
@@ -770,17 +856,36 @@ class BetterPlayerController {
     cancelNextVideoTimer();
   }
 
+  ///Setup track parameters for currently played video. Can be only used for HLS or DASH
+  ///data source.
+  void setTrack(dynamic track) {
```

Comment:
Maybe you could create BetterPlayerBaseTrack which will be both extended by Hls and Dash track class? I do not prefer dynamic here. For example user can pass here String which will trigger setDashTrack with String parameter which will trigger exception.

Reply:
I made this to maintain backward compatibility with current version. The best way should be implement the asms (adaptive streaming media sources) class and inherit from it HLS and DASH classes.
If it's ok for you, where should you place this file? in core folder or in another folder like asms?

Reply:
Implemented in a81bf01